Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py

```diff
@@ -1899,6 +1900,40 @@ def cancel_action(self, action_id, request_body):
         response = self.ms_client.http_request(method='POST', url_suffix=cmd_url, json_data=request_body)
         return response
 
+    def get_machine_users(self, machine_id):
+        """Retrieves a collection of users related to a given machine ID (logon users).
+        https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/get-machine-log-on-users?view=o365-worldwide
+
+        Args:
+            machine_id (str): The machine ID
+
+        Returns:
+            dict. User entities
+        """
+        cmd_url = f"/machines/{machine_id}/logonusers"
+        try:
+            response = self.ms_client.http_request(method="GET", url_suffix=cmd_url)
+        except Exception:
+            raise Exception(f"Machine {machine_id} was not found")
```

Comment:
Is this the only exception that can be raised from that command?

Reply:
Follow [Microsoft Documentation](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/get-machine-log-on-users?view=o365-worldwide) it is: 

> If successful and device exists - 200 OK with list of [user](https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/user?view=o365-worldwide) entities in the body. If device was not found - 404 Not Found.